go-controller/pkg/ovn/acl.go

```diff
@@ -0,0 +1,89 @@
+package ovn
+
+import (
+	"fmt"
+
+	libovsdbclient "github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdbops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/types"
+
+	"k8s.io/klog/v2"
+)
+
+// GetNamespaceACLLogging retrieves ACLLoggingLevels for the Namespace.
+// nsInfo will be locked (and unlocked at the end) for given namespace if it exists.
+func (oc *Controller) GetNamespaceACLLogging(ns string) *ACLLoggingLevels {
+	nsInfo, nsUnlock := oc.getNamespaceLocked(ns, true)
+	if nsInfo == nil {
+		return &ACLLoggingLevels{
+			Allow: "",
+			Deny:  "",
+		}
+	}
+	defer nsUnlock()
+	return &nsInfo.aclLogging
+}
+
+func getLogSeverity(action string, aclLogging *ACLLoggingLevels) (log bool, severity string) {
+	severity = ""
+	if aclLogging != nil {
+		if action == nbdb.ACLActionAllow || action == nbdb.ACLActionAllowRelated || action == nbdb.ACLActionAllowStateless {
+			severity = aclLogging.Allow
+		} else if action == nbdb.ACLActionDrop || action == nbdb.ACLActionReject {
+			severity = aclLogging.Deny
+		}
+	}
+	log = severity != ""
+	return log, severity
+}
+
+// SetACLLogging returns log and severity values that should be passed to
+func SetACLLogging(acl *nbdb.ACL, aclLogging *ACLLoggingLevels) {
+	log, severity := getLogSeverity(acl.Action, aclLogging)
+	libovsdbops.SetACLLogging(acl, severity, log)
+}
+
+// UpdateACLLoggingWithPredicate finds all ACLs based on a given predicate, updates log settings,
+// then transacts these changes with a single transaction.
+func UpdateACLLoggingWithPredicate(nbClient libovsdbclient.Client, p func(*nbdb.ACL) bool, aclLogging *ACLLoggingLevels) error {
+	ACLs, err := libovsdbops.FindACLsWithPredicate(nbClient, p)
+	if err != nil {
+		return fmt.Errorf("unable to list ACLs with predicate, err: %v", err)
+	}
+	if len(ACLs) == 0 {
+		klog.Warningf("No ACLs to update")
+		return nil
+	}
+	for i := range ACLs {
+		SetACLLogging(ACLs[i], aclLogging)
+	}
+	ops, err := libovsdbops.UpdateACLsLoggingOps(nbClient, nil, ACLs...)
+	if err != nil {
+		return fmt.Errorf("unable to get ACL logging ops: %v", err)
+	}
+	if _, err := libovsdbops.TransactAndCheck(nbClient, ops); err != nil {
+		return fmt.Errorf("unable to update ACL logging: %v", err)
+	}
+	return nil
+}
```

Comment:
This does seem like it belongs to libovsdbops/acl.go

Reply:
it uses `ACLLoggingLevels` and `SetACLLogging` from ovn package, that's mostly why ovn/acl exists